RecoLocalTracker/SiPhase2VectorHitBuilder/src/VectorHitBuilderAlgorithm.cc

```diff
@@ -0,0 +1,553 @@
+  if (theUpperDetSet.size() > 1)
+    LogDebug("VectorHitBuilderAlgorithm") << " more than 1 upper cluster! " << std::endl;
+  for (const_iterator cil = theLowerDetSet.begin(); cil != theLowerDetSet.end(); ++cil) {
+    Phase2TrackerCluster1DRef clusterLower = edmNew::makeRefTo(clusters, cil);
+    lowerClusters.push_back(*clusterLower);
+  }
+  return true;
+}
+
+bool VectorHitBuilderAlgorithm::checkClustersCompatibility(Local3DPoint& poslower,
+                                                           Local3DPoint& posupper,
+                                                           LocalError& errlower,
+                                                           LocalError& errupper) const {
+  return true;
+}
+
+//----------------------------------------------------------------------------
+//ERICA::in the DT code the global position is used to compute the alpha angle and put a cut on that.
+std::vector<std::pair<VectorHit, bool>> VectorHitBuilderAlgorithm::buildVectorHits(
+    const StackGeomDet* stack,
+    edm::Handle<edmNew::DetSetVector<Phase2TrackerCluster1D>> clusters,
+    const detset& theLowerDetSet,
+    const detset& theUpperDetSet,
+    const std::vector<bool>& phase2OTClustersToSkip) const {
+  std::vector<std::pair<VectorHit, bool>> result;
+  if (checkClustersCompatibilityBeforeBuilding(clusters, theLowerDetSet, theUpperDetSet)) {
+    LogDebug("VectorHitBuilderAlgorithm") << "  compatible -> continue ... " << std::endl;
+  } else {
+    LogTrace("VectorHitBuilderAlgorithm") << "  not compatible, going to the next cluster";
+  }
+
+  unsigned int layerStack = tkTopo_->layer(stack->geographicalId());
+  if (stack->subDetector() == GeomDetEnumerators::SubDetector::P2OTB)
+    LogDebug("VectorHitBuilderAlgorithm") << " \t is barrel.    " << std::endl;
+  if (stack->subDetector() == GeomDetEnumerators::SubDetector::P2OTEC)
+    LogDebug("VectorHitBuilderAlgorithm") << " \t is endcap.    " << std::endl;
+  LogDebug("VectorHitBuilderAlgorithm") << " \t layer is : " << layerStack << std::endl;
+
+  float cut = 0.0;
+  if (stack->subDetector() == GeomDetEnumerators::SubDetector::P2OTB)
+    cut = barrelCut_.at(layerStack);
+  if (stack->subDetector() == GeomDetEnumerators::SubDetector::P2OTEC)
+    cut = endcapCut_.at(layerStack);
+  LogDebug("VectorHitBuilderAlgorithm") << " \t the cut is:" << cut << std::endl;
+
+  //only cache local parameters for upper cluster as we loop over lower clusters only once anyway
+  std::vector<std::pair<LocalPoint, LocalError>> localParamsUpper;
+  std::vector<const PixelGeomDetUnit*> localGDUUpper;
+
+  std::vector<Phase2TrackerCluster1DRef> upperClusters;
+  const PixelGeomDetUnit* gduUpp = dynamic_cast<const PixelGeomDetUnit*>(stack->upperDet());
+  for (auto const& clusterUpper : theUpperDetSet) {
+    localGDUUpper.push_back(gduUpp);
+    localParamsUpper.push_back(cpe_->localParameters(clusterUpper, *gduUpp));
+  }
+  int upperIterator = 0;
```

Comment:
if we are still going to produce vectors, 
somewhere here it's worth to reserve the result to be the size of `theUpperDetSet` for the accepted part and if no-cut situation remains, reserve to `upper*lower` for the rejected (or combined) result

Reply:
No more vectors!